modules/api/api.py

```diff
@@ -515,6 +515,11 @@ def get_config(self):
 
     def set_config(self, req: Dict[str, Any]):
         for k, v in req.items():
+            if k == "sd_model_checkpoint":
+                checkpoint_info = checkpoint_alisases.get(v, None)
+                if checkpoint_info is None:
+                    print(f"model [{v}] not founded, skip config saving process")
+                    return 
```

Comment:
For a request like `{"foo": 1, "sd_model_checkpoint": "bar"}` (assuming `bar` does not exist), this will still first set `"foo"` to 1 in memory, then stop processing at a bad checkpoint, and finally not save anything.

You should probably validate first, only then act – and also, I don't think it's very good error handling to just print something on console and not inform the API caller about the problem?

Finally, this would simplify to

```suggestion
            if k == "sd_model_checkpoint":
                if v not in checkpoint_alisases:
                    raise RuntimeError(f"model {v!r} not found")
```